### PR TITLE
Bug 1876913: Configure proxy via git config

### PR DESCRIPTION
**Pull request by adambkaplan**
Configure HTTP/HTTPS proxy using git command line, rather than environment variables. This is a work-around for [BZ 1875639](https://bugzilla.redhat.com/show_bug.cgi?id=1875639).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,24 +10,23 @@ require (
 	github.com/containers/storage v1.20.2
 	github.com/docker/distribution v2.7.1+incompatible
 	github.com/elazarl/goproxy v0.0.0-20191011121108-aa519ddbe484 // indirect
-	github.com/evanphx/json-patch v4.5.0+incompatible // indirect
 	github.com/fsouza/go-dockerclient v1.6.3
 	github.com/opencontainers/runc v1.0.0-rc91.0.20200707015106-819fcc687efb
 	github.com/opencontainers/runtime-spec v1.0.3-0.20200520003142-237cc4f519e2
-	github.com/openshift/api v0.0.0-20200326160804-ecb9283fe820
-	github.com/openshift/client-go v0.0.0-20200326155132-2a6cd50aedd0
+	github.com/openshift/api v0.0.0-20200827090112-c05698d102cf
```
Comment (otaviof) [inline go.mod:16]: Are the changes in `go.mod` related to the "no proxy" issue?

Comment (adambkaplan) [reply go.mod:16]: This came in from the bump to library-go.

**Review by adambkaplan — COMMENTED** [review]
/assign @gabemontero 

/cc @coreydaley @otaviof

**Comment by gabemontero** [comment]
/lgtm

@adambkaplan presumably when this lands, in addition to the immediate concerns, in theory we've pulled in requisite git fixes to satisfy https://issues.redhat.com/browse/BUILD-68 as well

I would suggest we move that story to on-qe in our first 4.7 sprint for @wewang58 to try out

**Pull request merged**